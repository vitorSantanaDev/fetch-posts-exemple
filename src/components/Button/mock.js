export const buttonProps = {
  text: 'Load More Posts',
  onClick: jest.fn(),
  disabled: false,
};
