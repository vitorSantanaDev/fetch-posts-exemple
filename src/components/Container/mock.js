import React from 'react';

export const propsContainer = {
  children: <h1>Children</h1>,
};
