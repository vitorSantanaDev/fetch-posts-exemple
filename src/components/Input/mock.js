export const propsInput = {
  value: '',
  onChange: jest.fn(),
};
