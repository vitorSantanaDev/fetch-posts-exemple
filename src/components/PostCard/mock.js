export const postCardProps = {
  title: 'title one',
  body: 'Body one',
  cover: 'img/img.png',
};
