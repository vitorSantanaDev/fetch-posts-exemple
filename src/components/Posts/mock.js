export const postsProps = [
  { id: 1, title: 'Title Post one', body: 'Body text Post one', cover: '/image/image.png' },
  { id: 2, title: 'Title Post two', body: 'Body text Post two', cover: '/image/image.png' },
  { id: 3, title: 'Title Post three', body: 'Body text Post three', cover: '/image/image.png' },
];
