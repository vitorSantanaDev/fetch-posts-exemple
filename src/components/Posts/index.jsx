import React from 'react';
import './styles.css';
import PostCard from '../PostCard';
import Types from 'prop-types';

const Posts = ({ posts }) => {
  return (
    <div className="posts">
      {posts.map((post) => (
        <PostCard key={post.id} cover={post.cover} title={post.title} body={post.body} />
      ))}
    </div>
  );
};

Posts.propTypes = {
  posts: Types.array.isRequired,
};

export default Posts;
