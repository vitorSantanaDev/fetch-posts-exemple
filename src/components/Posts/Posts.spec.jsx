import { render, screen } from '@testing-library/react';
import Posts from '.';
import { postsProps } from './mock';

describe('<Posts />', () => {
  it('should render Posts', () => {
    render(<Posts posts={postsProps} />);
    expect(screen.getAllByRole('heading', { name: /title/i })).toHaveLength(3);
    expect(screen.getAllByRole('img', { name: /title/i })).toHaveLength(3);
    expect(screen.getAllByText(/body/i)).toHaveLength(3);
  });

  it('should not render Posts', () => {
    render(<Posts posts={[]} />);
    expect(screen.queryAllByRole('heading', { name: /title/i }));
  });

  it('should match snapshot', () => {
    const { container } = render(<Posts posts={postsProps} />);
    expect(container.firstChild).toMatchSnapshot();
  });
});
