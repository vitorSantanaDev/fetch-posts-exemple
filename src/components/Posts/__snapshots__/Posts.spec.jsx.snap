// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Posts /> should match snapshot 1`] = `
<div
  class="posts"
>
  <div
    class="post"
  >
    <img
      alt="Title Post one"
      src="/image/image.png"
    />
    <div
      class="postContent"
    >
      <h2>
        Title Post one
      </h2>
      <p>
        Body text Post one
      </p>
    </div>
  </div>
  <div
    class="post"
  >
    <img
      alt="Title Post two"
      src="/image/image.png"
    />
    <div
      class="postContent"
    >
      <h2>
        Title Post two
      </h2>
      <p>
        Body text Post two
      </p>
    </div>
  </div>
  <div
    class="post"
  >
    <img
      alt="Title Post three"
      src="/image/image.png"
    />
    <div
      class="postContent"
    >
      <h2>
        Title Post three
      </h2>
      <p>
        Body text Post three
      </p>
    </div>
  </div>
</div>
`;
